src/components/Personalization/createViewCacheManager.js

```diff
@@ -10,39 +10,84 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { assign } from "../../utils";
+import { assign, groupBy } from "../../utils";
 import defer from "../../utils/defer";
+import { DEFAULT_CONTENT_ITEM } from "./constants/schema";
 
-export default () => {
-  let viewStorage;
-  const viewStorageDeferred = defer();
+export default ({ createProposition }) => {
+  const viewStorage = {};
+  let cacheUpdateCreatedAtLeastOnce = false;
+  let previousUpdateCacheComplete = Promise.resolve();
 
-  const storeViews = decisionsPromise => {
-    decisionsPromise
-      .then(decisions => {
-        if (viewStorage === undefined) {
-          viewStorage = {};
-        }
-        assign(viewStorage, decisions);
-        viewStorageDeferred.resolve();
+  const getViewPropositions = (currentViewStorage, viewName) => {
+    const viewPropositions = currentViewStorage[viewName.toLowerCase()];
+    if (viewPropositions && viewPropositions.length > 0) {
+      return viewPropositions;
+    }
+
+    const emptyViewProposition = createProposition(
+      {
+        scope: viewName,
+        scopeDetails: {
+          characteristics: {
+            scopeType: "view"
+          }
+        },
+        items: [
+          {
```

Comment:
Can we remove the `Default Content Item` schema it will causes confusion. 
`Default Content Item` schema is used to notify client side that the user has entered a Target Activity but it doesn't have an HTML change to render.